STYLE_GUIDE.md

```diff
@@ -25,34 +30,18 @@ More generally when adding functions from external packages (i.e. even if they a
 
 ## Input types and checking
 
-- We support inputs that are coercible to `data.table` objects using `data.table::as.data.table()`. This includes `data.frame` and `tibble` objects. This should be clearly documented in the function documentation.
-- Any required inputs should be clearly documented in the function documentation.
-- We use an internal function `coerce_dt()` to check inputs are coercible to `data.table` objects and have the correct columns. This function is used in all functions that take data as input. The following function demonstrates this pattern (note this requires the use of `devtools::load_all()` in the package directory):
-
-```r
-print_dt <- function(dt) {
-    dt <- coerce_dt(dt, required_cols = c("date", "cases"))
-    return(dt[])
-}
-
-print_dt(mtcars)
-# Error in epinowcast:::coerce_dt(dt, required_cols = c("date", "cases")) : 
-#   The following columns are required: date, cases but are not present among mpg, cyl, disp, hp, drat, wt, qsec, vs, am, gear, carb
-# (all `required_cols`: date, cases)
-print_dt(data.frame(cases = 1, date = Sys.Date()))
-#    cases       date
-# 1:     1 2023-04-27
-```
-
-In general, we aim to check the inputs for all external facing functions. This is to ensure that the user is aware of any issues with the input data and to provide a consistent error message. See the documentation for `coerce_dt()` for more details. It may also be helpful to review usage in the package more widely, for this `data-converters.R` is a sensible place to start.
+- Any required inputs should be clearly documented in the function documentation, particularly in terms of type, but also other constraints (e.g. presence/absence of columns).
+- Any expressed constraint in exported functions should be verified using some sort of `check_` expression, and ideally unit tests written correspondingly to confirm that `check_` rejects bad input.
+- Many of methods across the `epinowcast` packages work with `data.frame` inputs (and subclasses of `data.frame` like `data.table` and `tibble`). Internally, for performance and syntax reasons, we prefer to use `data.table`s explicitly and that is generally the type returned by functions. However, we will continue to accept `data.frame`-like arguments as inputs.
+- Translation from `data.frame` to `data.table` can be handled by `coerce_dt`, `check_dt` or `make_dt` (`coerce_dt` and `check_dt` combined) which will be provided in a to-be-released package `make_dt`
```

Comment:
In other packages not written by us I think we should be open to a dplyr workflow. This sounds a little like we are not but I think we can probably cross that bridge when it comes.